xla/service/gpu/llvm_gpu_backend/amdgpu_backend.cc

```diff
@@ -171,10 +174,74 @@ void HsacoCache::Add(const std::string& ir, uint64_t hash,
   g_hsacoCache.cache.back().hsaco = hsaco;
 }
 
+
+void find_file(const std::string& root,
+               const std::string& filename,
+               std::vector<std::string>& results) {
+
+  auto* env = tsl::Env::Default();
+
+  std::vector<std::string> children;
```

Comment:
Do a vector of Paths instead. That way you dont have to convert things back and forth over and over.